instrumentation/resque/lib/opentelemetry/instrumentation/resque/patches/resque_job.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Resque
+      module Patches
+        # Module to prepend to Resque::Job for instrumentation
+        module ResqueJob
+          def perform # rubocop:disable Metrics/AbcSize, Metrics/MethodLength, Metrics/CyclomaticComplexity
+            job_class = payload_class_name
+
+            attributes = {
+              'messaging.system' => 'resque',
+              'messaging.destination' => queue.to_s,
+              'messaging.destination_kind' => 'queue',
+              'messaging.resque.job_class' => job_class
+            }
+
+            span_name = case config[:span_naming]
+                        when :queue then "#{queue} process"
+                        when :job_class then "#{job_class} process"
+                        else "#{queue} process"
+                        end
+
+            extracted_context = OpenTelemetry.propagation.extract(@payload)
+
+            OpenTelemetry::Context.with_current(extracted_context) do
+              if config[:propagation_style] == :child
+                tracer.in_span(span_name, attributes: attributes, kind: :consumer) { super }
+              else
+                links = []
+                span_context = OpenTelemetry::Trace.current_span(extracted_context).context
+                links << OpenTelemetry::Trace::Link.new(span_context) if config[:propagation_style] == :link && span_context.valid?
+                span = tracer.start_root_span(span_name, attributes: attributes, links: links, kind: :consumer)
+                OpenTelemetry::Trace.with_span(span) do
+                  super
+                rescue Exception => e # rubocop:disable Lint/RescueException
+                  span&.record_exception(e)
+                  span&.status = OpenTelemetry::Trace::Status.new(OpenTelemetry::Trace::Status::ERROR, description: "Unhandled exception of type: #{e.class}")
```

Comment:
```suggestion
                  span.record_exception(e)
                  span.status = OpenTelemetry::Trace::Status.new(OpenTelemetry::Trace::Status::ERROR, description: "Unhandled exception of type: #{e.class}")
```
I don't think `span` can be `nil` here.